letter-sender/src/test/java/com/christmas/letter/sender/controller/LetterReceiverControllerIntegrationTest.java

```diff
@@ -0,0 +1,101 @@
+package com.christmas.letter.sender.controller;
+
+import com.christmas.letter.sender.dto.ChristmasLetter;
+import com.christmas.letter.sender.service.LetterSenderService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.testcontainers.shaded.com.fasterxml.jackson.core.JsonProcessingException;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+import software.amazon.awssdk.services.sns.model.NotFoundException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.hasItem;
+import static org.mockito.Mockito.doThrow;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest
+public class LetterReceiverControllerIntegrationTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private LetterSenderService letterSenderService;
+
+    @Test
+    public void publishLetter_WhenBadRequest_ShouldReturnErrorResponse() throws Exception {
+        mockMvc.perform(post("/letter")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(getBadChristmasLetterRequest()))
+                .andExpect(status().isBadRequest())
+                .andExpect(jsonPath("$.status").value(400))
+                .andExpect(jsonPath("$.code").value("failed.request.body.validation"))
+                .andExpect(jsonPath("$.reasons").isArray())
+                .andExpect(jsonPath("$.reasons", hasItem("email.must.be.valid")))
+                .andExpect(jsonPath("$.reasons", hasItem("name.cannot.be.null")))
+                .andExpect(jsonPath("$.reasons", hasItem("name.cannot.be.empty")))
+                .andExpect(jsonPath("$.reasons", hasItem("must.add.at.least.one.wish")))
+                .andExpect(jsonPath("$.reasons", hasItem("address.cannot.be.null")))
+                .andExpect(jsonPath("$.reasons", hasItem("address.cannot.be.empty")));
+    }
+
+    @Test
+    public void publishLetter_WhenAWSSNSTopicNotFound_ShouldReturnErrorResponse() throws Exception {
+
+        ChristmasLetter requestBody = getChristmasLetterRequestJSON();
+        String requestBodyString = getChristmasLetterRequest();
+        doThrow(NotFoundException.class).when(letterSenderService).publishLetter(requestBody);
```

Comment:
If the request body is irelevant, you can simply use 
        doThrow(NotFoundException.class).when(letterSenderService).publishLetter(any(ChristmasLetter.class));

but... this isn't a scenario we should test. Yes, the application should fail, but we shouldn't have code written for this.